Updating and adding more information about the cht-core web applications (#438)
medic/cht-docs: content/en/core/overview/architecture.md

### cht-core

[cht-core](https://github.com/medic/cht-core) is composed of the following web applications:

> #### CHT Web Application

Author: I struggled a bit with webapp name, so instead of "webapp" I used "CHT Web Application" because I noticed that "admin" is called "App Management" in some other places. Is it okay?

> The CHT Web Application is the main entry point for must users, providing a large variety of [features](https://docs.communityhealthtoolkit.org/apps/features/).

Author: Done!

> Click [here](https://github.com/medic/cht-core/tree/master/webapp) to see the application source code.

Author: Done

##### Technology
The CHT Web Application is [reactive](https://angular.io/guide/rx-library), responsive and a single page application built with [Angular](https://angular.io/) and [NgRx](https://ngrx.io) frameworks.

> It uses an offline first strategy which means the data is stored on the client and all pages can load immediately regardless of whether the user has a fast connection, slow connection, or no connection at all. The data is stored in [PouchDB](https://pouchdb.com) which replicates changes back and forth in the background with the server CouchDB.

Author: Here I tried to follow what we had before: "It uses [tech] for [reason]"
But I think a table or bullets would look cleaner, for example:

- [Enketo](https://enketo.org): Used to render...

It uses [Enketo](https://enketo.org) to render configured xforms and help with styling and dynamic elements such as show/hide and validation rules.

It uses the [Nools](https://github.com/C2FO/nools) rules engine to compute the upcoming tasks and monthly targets of the users.

It uses the [Ngx-Bootstrap](https://github.com/valor-software/ngx-bootstrap) to integrate Bootstrap components in the Angular application.

> It uses the [Ngx-translate](https://github.com/ngx-translate/core) for Angular framework to automatically translate the application labels. See [here](https://docs.communityhealthtoolkit.org/apps/reference/translations/) how to manage translations.

Reviewer: ```suggestion
It uses the [Ngx-translate](https://github.com/ngx-translate/core) for Angular framework to automatically translate the application labels. Read more about [how to configure translations](https://docs.communityhealthtoolkit.org/apps/reference/translations/).
```

It uses [Protractor](https://www.protractortest.org/#/) to run the [application’s e2e tests](https://github.com/medic/cht-core/tree/master/tests/e2e)

It uses [Less](http://lesscss.org/) as a CSS preprocessor.

##### Structure

The CHT Web Application has the following high level structure: 

- **/js**: Contains the vanilla JavaScript scripts, for example: Enketo widgets, MomentJS locales, etc.
- **/ts**: Contains the Angular application source code which uses TypeScript.
  - **/actions**, **/effects**, **/reducers** and **/selectors**: Contain the implementation for the application’s reactive state which uses [NgRx](https://ngrx.io) framework.
  - **/components**, **/directives**, **/pipes**, **/providers** and **/services**: Contain the reusable elements from [Angular](https://angular.io/) framework.
  - **/modals**: Contains the all application’s modals components.
  - **/modules**: Contains the application’s modules, each of them has components that are associated to the modules’ routing.
- **/css**:  Contains the application’s style files. It uses [Less](http://lesscss.org/) as a CSS preprocessor. 
- **/fonts**: Contains the application fonts.

> - **/img**: Contains the application static images.

Author: Nice!

#### App Management
[App Management](https://docs.communityhealthtoolkit.org/apps/features/admin/) is an interface for non-technical administrative users to manage users and settings.

Click [here](https://github.com/medic/cht-core/tree/master/admin) to see the application source code.

##### Technology
App Management is a single page application built with [AngularJS](https://angularjs.org) framework and implements [Redux](https://github.com/reduxjs/redux) to manage a reactive state.

> It uses [PouchDB](https://pouchdb.com) to store the data and it replicates changes back and forth in the background with the server CouchDB.

Reviewer: Mostly it communicates directly with CouchDB as the admin users are online only. I'd just remove this line.

Author: Done

> It uses [Angular Translate](https://github.com/angular-translate/angular-translate) for AngularJS to automatically translate the application labels. See [here](https://docs.communityhealthtoolkit.org/apps/reference/translations/) how to manage translations.  

Reviewer: ```suggestion
It uses [Angular Translate](https://github.com/angular-translate/angular-translate) for AngularJS to automatically translate the application labels. Read more about [how to manage translations](https://docs.communityhealthtoolkit.org/apps/reference/translations/).  
```

> It uses [Karma](https://github.com/karma-runner/karma) and [MochaJS](https://mochajs.org/) to run the [application’s unit tests](https://github.com/medic/cht-core/tree/master/admin/tests)

Reviewer: ```suggestion
It uses [Karma](https://github.com/karma-runner/karma) and [MochaJS](https://mochajs.org/) to run the [unit tests](https://github.com/medic/cht-core/tree/master/admin/tests)
```

> It uses [Protractor](https://www.protractortest.org/#/) to run the [application’s e2e tests](https://github.com/medic/cht-core/tree/master/tests/e2e)

Reviewer: ```suggestion
It uses [Protractor](https://www.protractortest.org/#/) to run the [e2e tests](https://github.com/medic/cht-core/tree/master/tests/e2e)
```

> It uses [Less](http://lesscss.org/) as a CSS preprocessor.

Reviewer: It's probably worth using a table here as well so it's consistent with the above.

Author: Done

##### Structure

> - **/css**: Contains the application’s style files. It uses [Less](http://lesscss.org/) as a CSS preprocessor. 
> - **/js**: Contains the application JavaScript code. 
>   - **/actions**, **/reducers** and **/selectors**: Contain the implementation of [Redux](https://github.com/reduxjs/redux). 
>   - **/controllers**, **/directives**, **/filters** and **/services**: Contain the reusable elements from [AngularJS](https://angularjs.org) framework.
>   - **/modules**: Contains the vanilla JavaScript scripts.
> - **/template**: Contains the application’s HTML templates that are used in the AngularJS components and directives.

Reviewer: ```suggestion
- **/css**: Contains the style files. It uses [Less](http://lesscss.org/) as a CSS preprocessor. 
- **/js**: Contains the JavaScript code. 
  - **/actions**, **/reducers** and **/selectors**: Contain the implementation of [Redux](https://github.com/reduxjs/redux). 
  - **/controllers**, **/directives**, **/filters** and **/services**: Contain the reusable elements from [AngularJS](https://angularjs.org) framework.
  - **/modules**: Contains the vanilla JavaScript scripts.
- **/template**: Contains the HTML templates that are used in the AngularJS components and directives.
```
